core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1659,9 +1659,14 @@ private[spark] object Utils extends Logging {
   val windowsDrive = "([a-zA-Z])".r
 
   /**
-   * Format a Windows path such that it can be safely passed to a URI.
+   * Format a path such that it can be safely passed to a URI.
    */
-  def formatWindowsPath(path: String): String = path.replace("\\", "/")
+  def formatPath(path: String, windows: Boolean): String = {
+    val formatted = path.replace(" ", "%20")
```

Comment:
Do you mean we should assume `path` never contains fragment?
If we execute `bin/spark-shell --jars hdfs:/path/to/jar1.jar`, URI-style string is passed to `path`.
So I thought `path` may have fragment. (though I wonder if there are any good working example of the path with fragment...)